Plugins/VspSourceService/ViperArchiveSource/CMakeLists.txt

```diff
@@ -5,6 +5,9 @@ include_directories(SYSTEM
 )
 
 set(vsViperArchiveSource_Sources
+#moc was not being done on this file for the signals on Mac
+  vsViperArchiveSourcePrivate.h
```

Comment:
No other private headers, even those that are `QObject`s, need this. I would strongly prefer to know why and to avoid this one header being a special flower. (Note: I'd *thought* the usual solution was an empty source file of the same name, but I can't seem to find any of those in ViViA?)

IOW, why is this a problem for *this* header but not for:

- Applications/VsPlay/Plugins/vsVvqsDatabaseSourcePrivate.h
- Applications/Viqui/vqPredefinedQueryCachePrivate.h
- Plugins/VspSourceService/FakeStreamSourceFactory/vsFakeStreamSourcePrivate.h
- Libraries/VspUserInterface/vsMainWindowPrivate.h
- Libraries/VgVtkVideo/vgVideoPlayerPrivate.h
- Libraries/VgDataFramework/vdfTrackReaderPrivate.h
- Libraries/VspData/vsVideoSourcePrivate.h
- Libraries/VspData/vsLiveDescriptorPrivate.h
- Libraries/VspSourceUtil/vsStreamSourcePrivate.h

Reply:
...and the answer appears to be:

```diff
diff --git a/Plugins/VspSourceService/ViperArchiveSource/vsViperArchiveSource.cxx b/Plugins/VspSourceService/ViperArchiveSource/vsViperArchiveSource.cxx
index 6806fe1..5b83e73 100644
--- a/Plugins/VspSourceService/ViperArchiveSource/vsViperArchiveSource.cxx
+++ b/Plugins/VspSourceService/ViperArchiveSource/vsViperArchiveSource.cxx
@@ -4,7 +4,7 @@
  * Kitware, Inc., 28 Corporate Drive, Clifton Park, NY 12065.
  */
 
-#include "vsViperArchiveSourcePrivate.h"
+#include "moc_vsViperArchiveSourcePrivate.cpp"
 
 #include <QApplication>
 #include <QTimerEvent>
```

Please do that instead.

Reply:
This worked.